automation_tools/__init__.py

```diff
@@ -444,6 +444,42 @@ def setup_default_libvirt(bridge=None, ip_address="192.168.100.1"):
     return interface
 
 
+def setup_http_proxy():
+    """Postinstall task to set http_proxy and content_default_http_proxy settings for
+    Satellite >= 6.7
+
+    Proxy information is passed using the PROXY_INFO environmental variable.
+    The expected format is::
+
+        PROXY_INFO=proxy://<username>:<password>@<hostname>:<port>
+
+    ``username`` and ``password`` fields can be omitted if the proxy does not
+    require authentication, for example::
+
+        PROXY_INFO=proxy://<hostname>:<port>
+    """
+    proxy_info = os.environ.get('PROXY_INFO')
+    if proxy_info is None:
+        print('The PROXY_INFO environment variable should be defined')
+        sys.exit(1)
+    command = (
+        'hammer -u admin -p {0} settings set --name "http_proxy" --value {1}'
+    ).format(os.environ.get('ADMIN_PASSWORD', 'changeme'), proxy_info.replace('proxy:', 'http:'))
+    run(command, warn_only=True)
+    proxy_info = urlsplit(proxy_info)
+    url = 'http://{0}:{1}'.format(proxy_info.hostname, proxy_info.port)
+    creds = '--username {0} --password {1}'.format(proxy_info.username, proxy_info.password)
```

Comment:
Will these be always defined? I'm worried it will break if username or password ends up empty.

Reply:
@mirekdlugosz thanks for noticing that. I have updated the PR accordingly :D